[DEV] Add typecheck to lint workflow

flx-sta opened the pull request:
## What are the changes the user will see?

n/a

## Why am I making these changes?

To include `typecheck` in the lint workflow

## What are the changes from a developer perspective?

* **Add TypeScript type-checking step**: Introduced a new step in the GitHub Actions workflow to run `npm run typecheck`, ensuring TypeScript types are validated during the CI process. (`.github/workflows/lint.yml`, [.github/workflows/lint.ymlR55-R96](diffhunk://#diff-107e910e9f2ebfb9a741fa10b2aa7100cc1fc4f5f3aca2dfe78b905cbd73c0d2R55-R96))
* **Improve output formatting**: Replaced plain echo statements with a color-coded output using ANSI color codes to clearly indicate the success (`✅`) or failure (`❌`) of each check, including Biome, ESLint, Depcruise, and the new 
TypeScript type-check
* Updated README.md and linting.md to mention Biome

## Screenshots/Videos

See [Checks](https://github.com/Despair-Games/poketernity/pull/1210/checks) tab

## How to test the changes?

See [Checks](https://github.com/Despair-Games/poketernity/pull/1210/checks) tab

## Checklist

- [x] Otherwise: **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes manually?
- [x] Are all unit tests still passing? (`npm run test:silent`)

flx-sta committed:
Add typecheck to lint workflow

And enhance output

flx-sta committed:
update lint.yml

flx-sta committed:
Update lint.yml

MokaStitcher commented:
Seems like a good PR to update the linting information in the readme (there's nothing for biome)

flx-sta commented:
Todo: update linting info in readme

flx-sta committed:
update README.md

- add infos about Biome-Lint
- add infos about Dependency-Cruiser
- Fix formatting and headline levels

flx-sta committed:
add Biome info to linting.md

DayKev commented:
The typedoc ("GitHub Pages") workflow already handles typechecking. I'm not sure how exactly, though likely it just runs `tsc --no-emit` before it starts building the docs (based on some minimal investigation).

flx-sta commented:
> The typedoc ("GitHub Pages") workflow already handles typechecking. I'm not sure how exactly, though likely it just runs `tsc --no-emit` before it starts building the docs (based on some minimal investigation).

I know, but I feel it’s a bit of a hidden lint and should be rather Part of the actual linting flow

flx-sta committed:
Merge branch 'beta' into dev/add-typecheck-to-lint

MokaStitcher commented on README.md:
```diff
@@ -44,26 +44,42 @@ We are using [Prettier](https://prettier.io/) to format our code. It will run au
-#### Localization
+We are _still_ using [ESLint](https://eslint.org/docs/latest/rules/) plus the [ESLint Stylistic](https://eslint.style/rules) and [Typescript ESLint](https://typescript-eslint.io/rules/) plugins for linting. It will run automatically via the pre-commit hook, but if you would like to run it manually you can use the `npm run eslint` script. To view the currently applied ESLint rules, check out the [ ...
+
+#### Biome-Lint
```
```suggestion
##### Biome-Lint
```

MokaStitcher commented on README.md:
```diff
@@ -44,26 +44,42 @@ We are using [Prettier](https://prettier.io/) to format our code. It will run au
+> [!NOTE]
+> _We still require ESLint as some rules aren't covered by Biome yet. The plan is to migrate fully to Biome in the long run._
+
+#### Dependency-Cruiser
```
```suggestion
##### Dependency-Cruiser
```

MokaStitcher commented on README.md:
```diff
@@ -44,26 +44,42 @@ We are using [Prettier](https://prettier.io/) to format our code. It will run au
+> [!TIP]
+> Use `npm run depcruise` to check the codebase for any runtime circular dependency.
+
+## 🌐 Localization
```
```suggestion
#### 🌐 Localization
```
I think it makes sense to keep this under development guidelines since it's basically linking to the doc on how to make PRs with localisation

flx-sta committed:
Merge branch 'beta' into dev/add-typecheck-to-lint

flx-sta committed:
Update README.md

Co-authored-by: Moka <54149968+MokaStitcher@users.noreply.github.com>

flx-sta committed:
Update README.md

Co-authored-by: Moka <54149968+MokaStitcher@users.noreply.github.com>

flx-sta committed:
Update README.md

Co-authored-by: Moka <54149968+MokaStitcher@users.noreply.github.com>

flx-sta committed:
Merge branch 'beta' into dev/add-typecheck-to-lint

flx-sta committed:
Merge branch 'beta' into dev/add-typecheck-to-lint

flx-sta committed:
Merge branch 'beta' into dev/add-typecheck-to-lint

MokaStitcher commented on docs/linting.md:
```diff
@@ -42,3 +42,10 @@
+
+# Biome
+
+We started transitioning over to [Biome](https://biomejs.dev/) for linting as it's performance far exceeds the performance of ESLint (~15x faster).
+
+The idea is to have Biome replace ESLint completely in the future, but not all rules are supported yet so for now we're using both/a hybrid approach.
```
```suggestion
We started transitioning over to [Biome](https://biomejs.dev/) for linting as its performance far exceeds ESLint's (~15x faster).

The end goal is to have Biome replace ESLint completely in the future, but not all rules are supported by Biome yet so for now we're using a hybrid approach.
```

flx-sta committed:
Update docs/linting.md

Co-authored-by: Moka <54149968+MokaStitcher@users.noreply.github.com>
